scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
+        maximum_hit = False
+        if any(label['id'] == 'needs_review_reply_label' or label['name'] == 'needs_review_reply_label' for label in prdata['labels']['nodes']):
+          numbers.append(pr_number)
+        if prdata['closingIssuesReferences']['totalCount'] > 10 or prdata['labels']['totalCount'] > 10:
+          maximum_hit = True
+        for relatedIssue in prdata['closingIssuesReferences']['nodes']:
+          if any(label['id'] == 'needs_review_reply_label' for label in relatedIssue['labels']['nodes']):
+            numbers.append(relatedIssue['number'])
+          if relatedIssue['labels']['totalCount'] > 10:
+            maximum_hit = True
+        numbers_str = ', '.join(map(str, numbers))
+        issues_PRs_needing_review_replies.extend(numbers)
+        if maximum_hit:
+          numbers_str += " and more, possibly"
+          # TODO: move this whole block (which checks the linked issues of a PR and if they have the relevant label) to a separate function, which deals with pagination
+        if numbers_str:
+          needs_review_reply_string = f"Yes ({numbers_str})"
+        else:
+          needs_review_reply_string = "No"
+      
+      # Store pr information in list
+      prs_for_csv.append({
+        'mergedAt': prdata['mergedAt'],
+        'URL': prdata['url'],
+        'title': prdata['title'],
+        'author': prdata['author']['login'],
+        'relatedIssues': related_issues_string,
+        'labels': unique_labels_string,
+      })
+      if needs_review_reply_label:
+        prs_for_csv[-1]['needsReviewReplies'] = needs_review_reply_string
+
+  # Create a list of dictionaries with commits for in the CSV file
+  animation_state['text'] = "Clean up commit list"
+  ## Filter list with commit dictionaries so only ones without any associated PRs are left
+  commits = [commit for commit in commits if commit['pr_count'] == 0]
+
+  animation_state['text'] = "Get commit metadata"
+
+  ### Loop through commits to construct GraphQL query
+  query = f'''
+  query ($owner: String!, $repo: String!) {{
+    repository (name: $repo, owner: $owner) {{
+  '''
+
+  for n, commit in enumerate(commits):
+      query += f'''
+          commit{n}:object(oid: "{commit['sha']}") {{
+            ... on Commit {{
+              message
+              committedDate
+              url
+              authors(first:3) {{
+                totalCount
+                nodes {{
+                  user {{
+                    login
+                  }}
+                }}
+              }}
+            }}
+          }}
+  '''
+
+  query += f'''
+    }}
+  }}
+  '''
+  # Variable definition for GraphQL Explorer:
+  # {"sha": "863d4c3b611df83389d82958114bfd2d1204e457", "owner": "AntennaPod", "repo": "AntennaPod"}
+
+  ## Submit call
+  variables = {"owner": owner, "repo": repo}
+  result = client.execute(query, variables)
+  data = json.loads(result)
+
+  ## Parse response and add commit metadata to the list of commit dictionaries
```

Comment:
This feels a bit too generic to me. Shouldn't it be enough to just take the PR author? I know that theoretically someone else could open a PR where the commit author is someone else, but that never happens for AntennaPod. This just adds complexity that we don't really need.